tests/test_filter.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2018 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alberto Pérez García-Plaza <alpgarcia@bitergia.com>
+#
+
+import pandas
+import sys
+import unittest
+
+if '..' not in sys.path:
+    sys.path.insert(0, '..')
+
+from cereslib.dfutils.filter import FilterRows
+
+
+class TestFilter(unittest.TestCase):
+    """ Unit tests for Filter class
+    """
+
+    def test_filter_rows(self):
+        """ Test several cases for filtering rows by column value
+        """
+
+        # One column, values of different types
+        df = pandas.DataFrame()
+        filepaths = ['', None, '-', '/file/path', 1, True, pandas.np.nan, '-', [1, 2]]
+        df["filepath"] = filepaths
+        data_filtered = FilterRows(df)
+        df = data_filtered.filter_(["filepath"], "-")
+
+        self.assertEqual(len(df), 7)
+
+        # One empty column
+        df = pandas.DataFrame()
+        df["filepath"] = []
+        data_filtered = FilterRows(df)
+        df = data_filtered.filter_(["filepath"], "-")
+
+        self.assertEqual(len(df), 0)
+
+        # Several columns and just one empty
+        df = pandas.DataFrame()
+        df["filepath"] = []
+        df["name"] = ["name", "-", "other", "-"]
+        df["dirname"] = ["dir", "-", "-", "-"]
+        data_filtered = FilterRows(df)
+        df = data_filtered.filter_(["filepath", "name", "dirname"], "-")
+
+        self.assertEqual(len(df), 1)
+
+    def test_column_not_exists(self):
+        """ Test empty dataframe looking for the corresponding ValueError exception
+        """
+        df = pandas.DataFrame()
+        data_filtered = FilterRows(df)
+        with self.assertRaises(ValueError) as context:
+            data_filtered.filter_(["filepath"], "-")
+
+        self.assertTrue('Column filepath not in DataFrame columns: []' in str(context.exception))
```

Comment:
@alpgarcia it seems that using assertRaisesRegexp [1] is a better option than comparing strings in the way you did.

I've tested the following line and it seems to work for me. Please replace lines 71-74 by the following two lines and that should be enough:

with self.assertRaisesRegex(ValueError, "Column filepath not in DataFrame columns"):
    data_filtered.filter_(["filepath"], "-")


Thanks to @sduenas for his advice on this.

And thanks @alpgarcia for the PR!


[1] https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex